plugins/woocommerce/tests/e2e-pw/tests/merchant/lost-password.spec.js

```diff
@@ -20,19 +20,19 @@ test.describe( 'Can go to lost password page and submit the form', () => {
 		await page.getByRole( 'button', { name: 'Get New Password' } ).click();
 
 		try {
-			await page.waitForURL( '**/wp-login.php?checkemail=confirm' );
-			await expect(
-				page.getByText( /Check your email for the confirmation link/i )
-			).toBeVisible();
-		} catch ( e ) {
 			// For local testing, the email might not be sent, so we can ignore this error.
-
-			// eslint-disable-next-line jest/no-try-expect
 			await expect(
 				page.getByText(
 					/The email could not be sent. Your site may not be correctly configured to send emails/i
 				)
 			).toBeVisible();
```

Comment:
@chihsuan, we can use a `process.env.CI` variable to check against the current testing env. For example:

```js
if ( process.env.CI ) {
	await expect(
		page.getByText( /Check your email for the confirmation link/i )
	).toBeVisible();
} else {
	await expect(
		page.getByText(
			/The email could not be sent. Your site may not be correctly configured to send emails/i
		)
	).toBeVisible();
}
```

Reply:
Thanks for the suggestion, @WunderBart! 

Not sure if this is still applicable, but I think we don't want to care what's the end result, we just wanted to make sure that the button click attempts a password reset. I wasn't sure if it can be guaranteed to be one way or another. Could we say that in CI always, an email send always fails?

Reply:
Ah, this can be then simplified even further! We can use the [`locator.or`](https://playwright.dev/docs/api/class-locator#locator-or) API, for example:

```js
// The `getByText` locator matches a substring and is case-insensitive
// by default, so there is no need to use regexp here!
const emailSentMessage = page.getByText( 'check your email for the confirmation link' );
const emailNotSentMessage = page.getByText( 'the email could not be sent' );

await expect( emailSentMessage.or( emailNotSentMessage )).toBeVisible();
```

I'd recommend this approach as it's stable, handles errors gracefully, and does not introduce delays.

Reply:
Great suggestion, @WunderBart! TIL. I'll write a PR quick

Reply:
PR here: https://github.com/woocommerce/woocommerce/pull/50823